design/design.md

```diff
@@ -0,0 +1,79 @@
+
+We decided to have different emojis (🔥, ⚠️, 💤) for the different priority levels for people who are color blind. This was a suggestion from Prof. Rhodes. 
+
+We made the 💤 the default priority level for tasks because if a user does not bother to change the priority level of a task, it is usually a safe assumption that it is unimportant.
+
+## Color Palette
+
+We chose a very neutral grey color palette as our app will have a lot of color from the priority levels we can assign to each task and we do not want it to clash with the background and overall theme of the app.
+
+#ffffff, #e6e6e6, #999999, #494949
+
+### Dark Mode
+
+We also would like to implement a dark mode to invert the colors. Since our theme is greyscale, this should be easy to do without breaking the aesthetic of our app. 
+
+## Swipe Quick Actions
+
+This has not yet been implemented.
+
+We also would like to implement a quick swipe action for each task item similar to how mail apps allow users to quick mark an email as unread or to quick delete a mail by swiping that mail left/right. 
+
+We would like to implement quick swipe actions for quickly deleting the task and quickly renaming the task, respectively.
+
+We think that adding swipe quick actions can only be beneficial as to users who are not aware of its functionality, it does not take away from the user experience and does not clutter up the UI. It is invisible if the user chooses not to use it. 
+
+# Alternative Designs
+Initially, we had 4 buttons on the top bar: A clock, a stoplight, a checkmark, and the menu icon. 3 of these icons made it to the final design but the clock was scrapped. Initially, the clock and the stoplight were for ordering the task items by creation date and by priority level, respectively, and they would form a radio button group. However, we decided that sorting items by creation date wasn't very meaningful so we decided to just leave in the priority level stoplight button but as a singular toggle button instead.
+
+![initial design](initial_design.jpg)
+
+We also considered adding due date fields for each of the tasks, but we thought that might make the task too complicated and we were not sure how to display the information cleanly. This is still something we are looking into adding in the future.
+
+When sorting by priority level, we also considered having dividers within the task list itself to divide the tasks of high, medium, and low priority into their own sections. We did not have time to implement this but we will experiment with this to see if it makes the user experience better.
+
+![priority headers](priority_headers.jpg)
+
+# User Testing
+
+Initially, we wanted to not have labels associated with the menu icons on the top bar of our app, but from our user testing, none of the users we tested the product on were able to successfully guess what the icons represented or what their functionality was. Because of this, we decided to add labels to each of our icons on the top bar. 
```

Comment:
More information on user testing would be better: who did you test with, what were you testing and how?